R/checks.R

```diff
@@ -61,6 +61,7 @@ check_grid <- function(grid, workflow, pset = NULL) {
     }
 
     grid_distinct <- distinct(grid)
+    grid_distinct <- vctrs::vec_restore(grid_distinct, tibble::tibble())
```

Comment:
Where do these grid attributes come from?

If they come from a classed output of dials, then dials could provide an `as_tibble()` method for that class that stripped off all the attributes and the class.

Otherwise what I have here is probably the easiest alternative. It just creates a new bare tibble from the input, dropping all attributes except the names.

Reply:
Sigh - I forgot `new_tibble()` doesn't drop attributes either https://github.com/tidyverse/tibble/pull/769

So best to use `vctrs::new_data_frame()`, which works right. Then convert that to a tibble.

```r
library(tibble)
library(vctrs)
#> 
#> Attaching package: 'vctrs'
#> The following object is masked from 'package:tibble':
#> 
#>     data_frame

df <- tibble(x = 1)
attr(df, "foo") <- "bar"

# dammit tibble
df2 <- new_tibble(df, nrow = nrow(df))
attr(df2, "foo")
#> [1] "bar"

# ok vctrs does the right thing
df3 <- new_data_frame(df, n = nrow(df))
attr(df3, "foo")
#> NULL
as_tibble(df3)
#> # A tibble: 1 × 1
#>       x
#>   <dbl>
#> 1     1
```

We typically wrap this up in a helper like this and have it floating around in a few places

```r
new_tibble0 <- function(x, n) {
  out <- vctrs::new_data_frame(x, n = n)
  tibble::as_tibble(out)
}
```

Reply:
Got it, thanks for the feedback. :)

> Where do these grid attributes come from?

Wherever the user supplies them from, I think, though this piece of code is particularly written with an eye for removing attributes tacked on by `expand.grid()`.

Reply:
Fixed in [2acf1c5](https://github.com/tidymodels/tune/pull/605/commits/2acf1c54797ad698556a4c4fe07054b624c80aae)!